### PR TITLE
Bug OCPBUGS-828: backport `opm serve` caching functionality (and bugfix) + `opm serve` leaving orphan tmpfiles

### PR DESCRIPTION
**Pull request by grokspawn**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -234,7 +234,7 @@ require (
 	gomodules.xyz/jsonpatch/v2 v2.2.0 // indirect
 	google.golang.org/appengine v1.6.7 // indirect
 	google.golang.org/genproto v0.0.0-20220107163113-42d7afdf6368 // indirect
-	google.golang.org/grpc v1.43.0 // indirect
+	google.golang.org/grpc v1.45.0 // indirect
```
Comment (timflannagan) [inline go.mod:237]: It looks like we still replace pin this grpc package due to OLM issues correctly in the vendor/modules.txt

**Review by timflannagan — APPROVED** [review]

**Comment by awgreene** [comment]
This performance improvement needs to be backported because the default catalogs are constantly failing their readiness and liveness in 4.11 clusters.

/label backport-risk-assessed

**Comment by grokspawn** [comment]
/hold cancel
We have cherry-pick-approved, qe-approved, backport-risk-assessed

**Pull request merged**